### PR TITLE
Fix cmd-line help, add config option, use local lmmsrc file in dev

### LINKED ISSUE #2425: Add a command-line option to use a different resource file
When using various versions of Lmms on the same box you need different lmmsrc.xml files.
The goal is to implement a new command-line parameter to specify the resource file to use.
The default will stay, well, the default ~.lmmsrc.xml file but the command line -c or --config allow to overwrite it.

Also:
1- In master branch the --help is missing some already existing options:
--geometry
--import
--profile

2- There is a bug in --profile option: it takes <code>profilerOutputFile</code> from <code>argv[1]</code> instead of <code>argv[i]</code>

3- Could it be more clean to display the options in alphabetical  order in lmms --help?

I can fix these in the same issue or open separate ones for each.

### PR DESCRIPTION
**Pull request by midi-pascal**
Fix #2425 
- Add command-line option <code>--config \<configFile\></code> to specify a lmmsrc file instead of the default one
- Use lmmsrc file from the build directory in development mode (lmms not installed) instead of ~.lmmsrc.xml. This can be overriden by using the <code>--config</code> option described above.
- Update the command line <code>--help</code> to show all options available
- Fix the <code>--profile</code> option (the filename was taken from <code>argv[1]</code> instead of <code>argv[i]</code>
- Make the <code>-geometry</code> option <code>--geometry</code> (with two dashes) for consistency with other options

### CODE REVIEW

File: src/core/main.cpp [hunk 1]
```diff
@@ -116,39 +116,58 @@ void printHelp()
...
 	printf( "LMMS %s\n"
 		"Copyright (c) %s\n\n"
-		"Usage: lmms [ -r <project file> ] [ options ]\n"
-		"            [ -u <in> <out> ]\n"
+		"Usage: lmms [ -a ]\n"
+		"            [ -b <bitrate> ]\n"
+		"            [ -c <configfile> ]\n"
 		"            [ -d <in> ]\n"
+		"            [ -f <format> ]\n"
+		"            [ -geometry <geometry> ]\n"
```
Comment (Umcaruje) [inline src/core/main.cpp]: Should this here be `--geometry`?

Comment (midi-pascal) [inline src/core/main.cpp]: Good catch! Fixing...

**Comment by zonkmachine** [comment]
Trying it out now.

The help text will get scrambled if it isn't displayed at the same tab size as you had when you typed. Maybe convert the tabs to spaces?

```
-a, --float			32bit float bit depth
-b, --bitrate <bitrate>		Specify output bitrate in KBit/s
				Default: 160.
-c, --config <configfile>			Get the configuration from <configfile>
-d, --dump <in>			Dump XML of compressed file <in>
-f, --format <format>		Specify format of render-output where
				Format is either 'wav' or 'ogg'.
    --geometry <geometry>	Specify the prefered size and position of the main window
				geometry is <xsizexysize+xoffset+yoffsety>.
-h, --help			Show this usage information and exit.
-i, --interpolation <method>	Specify interpolation method
				Possible values:
				   - linear
```

**Comment by zonkmachine** [comment]
The man page needs the extra commands too.
https://github.com/LMMS/lmms/blob/master/doc/lmms.1

**Comment by midi-pascal** [comment]
@zonkmachine For the first point : I did not change the tab size in any way in the help screen but I can replace the tabs by 8 spaces if you think it is better...
For the second point : I am not familiar at all with the man page syntax but I can try - without any guaranty on the result :wink:

**Comment by midi-pascal** [comment]
@zonkmachine Ireplaced the tabs with spaces in help screen and updated the man page (it was not so difficult after all)

**Comment by zonkmachine** [comment]
Sorry for the lack of feedback.
 * main.cpp/menu. That's how I'd do it.
 * man page. Can't advise on it.
 * The separate .lmmsrc.xml is like... :+1: I applied the patch, entered `make`, and went for a cup of coffee. When I cam back it was like Richard Stallman had sneaked into my secret cave and sprinkled magic hacker dust over my hard drive...
I wonder if it makes compiling from source more complicated for beginners though?

**Comment by midi-pascal** [comment]
@zonkmachine 
To check the man page - if you run Linux - go to the doc directory and type <code>man -l lmms.1</code>. To me it looks good.
Sorry, I do not understand your third point since English is not my mother tongue... Does it work for you as expected or not?

**Comment by zonkmachine** [comment]
> To check the man page - if you run Linux - go to the doc directory and type man -l lmms.1. To me it looks good.

To me it looks OK too but I'm not a Debian/Ubuntu devotee. These people can be pretty particular about how things should be done.

> Sorry, I do not understand your third point since English is not my mother tongue... Does it work for you as expected or not?

C'est super!... :beers:

**Comment by midi-pascal** [comment]
<blockquote>To me it looks OK too but I'm not a Debian/Ubuntu devotee. These people can be pretty particular about how things should be done.</blockquote>
My mother tongue is Linux :wink: so if it is OK for you and for me, it should be OK for everyone...
Anyway I am not sure many people use the Lmms man page: before my update only 4 options on 19 were documented :-1:

**Comment by tresf** [comment]
FYI, `-geometry` is an [X11 throwback](http://www.tldp.org/HOWTO/XWindow-User-HOWTO/cli.html)... so it was likely intentional... let's keep both. :+1:

**Comment by midi-pascal** [comment]
@tresf Ok, I will do the fix to accept both options

**Comment by tresf** [comment]
@midi-pascal this looks ok.  Can you squash the commits and then we'll merge?

**Comment by Umcaruje** [comment]
Actually, github supports squashing right from the web interface, so we don't need to ask our contibutors to squash anymore: 
screenshot from 2016-04-23 23 17 09

**Comment by midi-pascal** [comment]
@Umcaruje :+1:  Nice!

**Pull request merged**